system-x/common/src/main/java/software/tnb/common/deployment/WithOperatorHub.java

```diff
@@ -0,0 +1,83 @@
+package software.tnb.common.deployment;
+
+import software.tnb.common.openshift.OpenshiftClient;
+import software.tnb.common.utils.WaitUtils;
+
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+
+public interface WithOperatorHub {
+    String operatorName();
+
+    default String operatorCatalog() {
+        return "redhat-operators";
+    }
+
+    default String operatorCatalogNamespace() {
+        return "openshift-marketplace";
+    }
+
+    default String operatorChannel() {
+        return "stable";
+    }
+
+    default String subscriptionName() {
+        return "tnb-" + getSuperClassName();
+    }
+
+    default String targetNamespace() {
+        return OpenshiftClient.get().getNamespace();
+    }
+
+    default String getOperatorCatalog() {
+        return getValue("operator.catalog", operatorCatalog());
+    }
+
+    default String getOperatorChannel() {
+        return getValue("operator.channel", operatorChannel());
+    }
+
+    default String getOperatorName() {
+        return getValue("operator.name", operatorName());
+    }
+
+    default String getOperatorCatalogNamespace() {
+        return getValue("catalog.namespace", operatorCatalogNamespace());
+    }
+
+    default boolean clusterWide() {
+        return false;
+    }
+
+    default void createSubscription() {
+        OpenshiftClient.get()
+            .createSubscription(getOperatorChannel(), getOperatorName(), getOperatorCatalog(), subscriptionName(), getOperatorCatalogNamespace(),
+                targetNamespace(), clusterWide());
+        OpenshiftClient.get().waitForInstallPlanToComplete(subscriptionName(), targetNamespace());
+    }
+
+    default void deleteSubscription(BooleanSupplier waitCondition) {
+        OpenshiftClient.get().deleteSubscription(subscriptionName(), targetNamespace());
+        WaitUtils.waitFor(waitCondition, "Waiting until the operator is undeployed");
+    }
+
+    /**
+     * This class should alwaye be OpenshiftX, so it should always extend some X class and the property to use
+     * to override the image would resolve to x.[property]
+     *
+     * @return value
```

Comment:
`alwaye` -> `always`
Not sure I understand what the doc is trying to say :D 
`so it should always extend some X class` -> classes should be named Openshift.*
`and the property to use to override the image would resolve to x.[property]` -> ?, is it related only to image? it should be any property specified in the property parameter no?

Reply:
ah I also now understand why the `x.[property]` is lowercase... maybe some concrete example would be more useful like 
OpenshiftMySQLServer -> mysqlserver.default-database=my-database